ext/datadog_profiling_native_extension/datadog_ruby_common.c

```diff
@@ -18,6 +21,14 @@ void raise_unexpected_type(VALUE value, const char *value_name, const char *type
   );
 }
 
+void raise_error(VALUE error_class, const char *fmt, ...) {
+  va_list args;
+  va_start(args, fmt);
+  VALUE message = rb_vsprintf(fmt, args);
+  va_end(args);
+  rb_raise(error_class, "%"PRIsVALUE, message);
+}
```

Comment:
Wait, I'm confused by this... Is this a leftover from an earlier version? 

That is, I thought the correct version of `raise_error` was the `#define raise_error` we have in `ruby_helpers.c` (the one that grabs the format string separately, etc); this one seems to just be a weird direct passthrough to `rb_raise`.

Am I missing something? 🤔 👀

Reply:
Nice catch!
This is left over from before.

I delete it and fixed the references, which created a small cascade of method movements.